apps/total-typescript/src/trpc/routers/search.ts

```diff
@@ -0,0 +1,56 @@
+import {publicProcedure, router} from '@skillrecordings/skill-lesson'
+import {z} from 'zod'
+import {NextApiRequest} from 'next'
+import {sanityClient} from '@skillrecordings/skill-lesson/utils/sanity-client'
+
+export const searchRouter = router({
+  resultsForQuery: publicProcedure
+    .input(
+      z.object({
+        query: z.string(),
+        numResults: z.number().default(20),
+      }),
+    )
+    .query(async ({ctx, input}) => {
+      const results = await sanityClient.fetch(
+        `  *[_type in ["article", "tip", "module", "exercise", "explainer"]]
+    | score(
+      title match $searchQuery 
+      || _type match "module"
+      || description match $searchQuery
+      || pt::text(body) match $searchQuery
+      || boost(pt::text(body) match $searchQuery + "*", 0.5)
+      || boost(pt::text(transcript) match $searchQuery, 0.5)
+
+    )
```

Comment:
this gives you a quick rank and seems to work very well.

Reply:
pretty cool!